test/functional/feature_coinstatsindex.py

```diff
@@ -247,6 +248,32 @@ def _test_use_index_option(self):
         del res['disk_size'], option_res['disk_size']
         assert_equal(res, option_res)
 
+    def _test_reorg_index(self):
+        self.log.info("Test that index can handle reorgs")
+
+        # Generate a block, let the index catch up, then invalidate the block
+        index_node = self.nodes[1]
+        reorg_block = index_node.generatetoaddress(1, index_node.getnewaddress())[0]
+        self.wait_until(lambda: not try_rpc(-32603, "Unable to read UTXO set", index_node.gettxoutsetinfo, 'muhash'))
+        index_node.invalidateblock(reorg_block)
+        assert_equal(index_node.gettxoutsetinfo('muhash')['height'], 110)
+
+        # Add two new blocks
+        blocks = index_node.generate(2)
+        self.wait_until(lambda: not try_rpc(-32603, "Unable to read UTXO set", index_node.gettxoutsetinfo, 'muhash'))
+
+        # Test that the result of the reorged block is not returned for its old block height
+        res = index_node.gettxoutsetinfo('muhash', 111)
+        assert_equal(res["bestblock"], blocks[0])
```

Comment:
Interesting. Thanks for the clearer description @Sjors. The reorg case that you describe here (querying a reorged out block by hash) would return an error until now because `ParseHashOrHeight()` is checking that the queried block is part of the active chain. But I think it could be interesting to have that functionality so I removed this check and added the test. It works fine. However, I am still a bit unsure if I should integrate this here or pull it out into a separate PR because `ParseHashOrHeight()` is also used by `getblockstats` so it leads to a behavior change there too. Tests are passing though and I tend to think it's ok. But I have left it as a separate WIP commit for now for easier inspection.

> Another thing you can test is what happens when you activate the index after a reorg (but I think fixing issues there can wait).

Not sure I fully understand this one. Do you mean just activating the index for the first time on a node that is aware of some stale blocks?

Reply:
> But I think it could be interesting to have that functionality so I removed this check and added the test. It works fine.

Nice!

> Do you mean just activating the index for the first time on a node that is aware of some stale blocks?

Correct. Do those stale blocks get indexed?

Reply:
Cool, I pulled in your changes and made you co-author. The stale blocks don't get indexed because the base index only follows the active chain when syncing. I have added some additional coverage for that in the reorg test part, mainly to document that behavior. I have pulled the change that allows querying stale blocks in its own commit so it's more clear that this also touches `getblockstats` and that that gets the attention such a side effect deserves.